core/src/main/scala/org/apache/spark/api/java/JavaSparkContext.scala

```diff
@@ -712,12 +712,51 @@ class JavaSparkContext(val sc: SparkContext) extends Closeable {
   /** Clear the current thread's job group ID and its description. */
   def clearJobGroup(): Unit = sc.clearJobGroup()
 
+  /**
+   * Set the behavior of job cancellation from jobs started in this thread.
+   *
+   * @param interruptOnCancel If true, then job cancellation will result in `Thread.interrupt()`
+   * being called on the job's executor threads. This is useful to help ensure that the tasks
+   * are actually stopped in a timely manner, but is off by default due to HDFS-1208, where HDFS
+   * may respond to Thread.interrupt() by marking nodes as dead.
+   */
+  def setInterruptOnCancel(interruptOnCancel: Boolean): Unit =
```

Comment:
shall we annotate them with `@since`

Reply:
Yeah, actually let me add them to Scala side since i am here.